src/plugins/field_formats/common/converters/duration.ts

```diff
@@ -128,53 +106,56 @@ export class DurationFormat extends FieldFormat {
   };
 }
 
+// function to calculate the precision part of the value
+const calculatePrecision = (valueInSeconds: number, unitValue: number, secondsPerUnit: number) => {
+  return Math.floor(unitValue) + (valueInSeconds - unitValue * secondsPerUnit) / secondsPerUnit;
+};
+
+// Array of units is to find the first unit duration value that is not 0
+const units = [
+  { getValue: (dur: moment.Duration) => dur.years(), seconds: 31536000, method: 'asYears' },
+  // Note: 30 days is used as a month in the duration format
+  { getValue: (dur: moment.Duration) => dur.months(), seconds: 2592000, method: 'asMonths' },
+  { getValue: (dur: moment.Duration) => dur.weeks(), seconds: 604800, method: 'asWeeks' },
```

Comment:
Using `weeks` in some cases is less convenient than using `days`:

<img width="458" alt="Screenshot 2024-04-29 at 09 58 17" src="https://github.com/elastic/kibana/assets/1415710/61ff4fb3-1422-4081-82ab-d3e7664ba275">

<img width="439" alt="Screenshot 2024-04-29 at 09 58 25" src="https://github.com/elastic/kibana/assets/1415710/2d5ef927-8904-4f81-b9e3-6536bd4c986e">

Anyway, it does not look like a blocker for the PR.

Reply:
So currently it works like this
if the duration in weeks is > 1  => weeks are displayed
if we would use 2 to check, days would be displayed until there are 14 days, then it would switch to weeks

ideally it would be 1 week and x days, but if we get down that path, it could also be 1 week x days x hours ...

so the question is, which unit - watershed to use for this case, so it's convenient? And are there more cases like this when different values should be used to determine with unit is should be displayed?

Reply:
I don't have a strong opinion on how it should be rendered. Maybe it's something to improve separately.

Reply:
yes, I don't think it needs to be in the scope of this PR, but it's a good remark, for sure